teuthology/suite/run.py

```diff
@@ -134,7 +136,7 @@ def choose_kernel(self):
             kernel_hash = None
         else:
             kernel_hash = util.get_gitbuilder_hash(
-                'kernel', self.args.kernel_branch, self.args.kernel_flavor,
+                'kernel', self.args.kernel_branch, self.flavor,
```

Comment:
The only kernel flavor we use today is `basic`.  It probably doesn't matter but just in case `basic` is still expected somewhere, I would suggest hard coding it here:
```suggestion
                'kernel', self.args.kernel_branch, 'basic',
```